File: lc_py/lcPy/prefix/checkSubarraySum.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        m = len(nums)
        mod_map = defaultdict(int)
        mod_map[0] = -1
        prefix_s = 0
        
        for i in range(m):
            prefix_s += nums[i]
            
            mod = prefix_s % k
            
            if mod in mod_map:
                if i - mod_map[mod] >= 2:
                    return True
            else:
                mod_map[mod] = i
        return False

File: lc_cpp/lcCpp/prefix/checkSubarraySum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        int m = nums.size();
        unordered_map<int, int> mod_map;
        mod_map[0] = -1;
        int prefix_s = 0;
        
        for (int i = 0; i < m; i++) {
            prefix_s += nums[i];
            
            int mod = prefix_s % k;
            
            if (mod_map.find(mod) != mod_map.end()) {
                if (i - mod_map[mod] >= 2) {
                    return true;
                }
            } else {
                mod_map[mod] = i;
            }
        }
        return false;
    }
};